sklearn/metrics/tests/test_common.py

```diff
@@ -1110,3 +1110,101 @@ def test_no_averaging_labels():
             score_labels = metric(y_true, y_pred, labels=labels, average=None)
             score = metric(y_true, y_pred, average=None)
             assert_array_equal(score_labels, score[inverse_labels])
+
+
+def test_metric_permutation_invariance_multiclass():
+    rng = np.random.RandomState(42)
+    n_classes = 5
+    n_samples = 100
+    p_C = rng.rand(n_classes)
+    p_C /= p_C.sum()
+    cp_C = np.cumsum(p_C)
+    y_true = np.searchsorted(cp_C, rng.rand(n_samples))
+    y_pred = []
+    for y in y_true:
+        if rng.randint(0, 10) > 5:
+            y_pred_new = (y + 1) % n_classes
+        else:
+            y_pred_new = y
+        y_pred.append(y_pred_new)
+    classes_perm = rng.permutation(n_classes)
+    y_true_perm = classes_perm[y_true]
+    y_pred_perm = classes_perm[y_pred]
+    for name in CLASSIFICATION_METRICS:
+        if name in METRIC_UNDEFINED_BINARY_MULTICLASS:
+            continue
+        if name is "confusion_matrix":
```

Comment:
One should not compare strings with `is`. The same string of characters may not have the same id:
```
>>> 'hello world'.split()[0] is 'hello'
False
```